cln_plugin/server.go

```diff
@@ -164,6 +167,11 @@ func (s *server) HtlcStream(stream proto.ClnPlugin_HtlcStreamServer) error {
 
 	s.htlcStream = stream
 
+	// Replay in-flight htlcs in fifo order
+	for pair := s.inflightHtlcs.Oldest(); pair != nil; pair = pair.Next() {
+		sendHtlcAccepted(stream, pair.Value)
```

Comment:
It seems here that we ignore errors. The main concern is that we will hold this htlc forever (causing the channel to be closed) in case of error as there is no retry logic like in the ongoing htlc accepted hook.

Reply:
The timeout listener is still running. So if this htlc is not handled, it will time out.

The ongoing htlc accepted hook waits for a new subscriber to send the message to.
This replay happens at the moment when lspd subscribes to the stream. So if the connection breaks, lspd will subscribe again and this part will be run again. My assumption was an error means the stream is broken.

I do agree on error we should break this loop. Also I noticed that the retry mechanism in combination with this replay will send the same htlc to lspd twice on reconnect. I'll fix that.

Reply:
Actually, we're not tracking the timeout anymore when the htlc is already sent. Good catch.

Reply:
This should be fixed now in the latest commit: https://github.com/breez/lspd/pull/125/commits/efec935ca2459b056ca498de8f1b1bc7cbe1fc39